DisplaySettings/DisplaySettings.cpp

```diff
@@ -850,9 +905,50 @@ namespace WPEFramework {
                 LOG_DEVICE_EXCEPTION1(zoomSetting);
                 success = false;
             }
+#endif
             returnResponse(success);
         }
 
+#ifdef ENABLE_TV_ZOOM_SETTINGS
+        bool DisplaySettings::setTVZoomSetting(std::string zoomSetting)
+        {
+            bool success = true;
+            if (std::find(tvZoomSettings.begin(), tvZoomSettings.end(), zoomSetting) != tvZoomSettings.end())
+            {
+                Core::File settingsFile;
+                settingsFile = TV_ZOOM_SETTINGS_FILE;
+                if (!settingsFile.Open(false))
+                {
+                    LOGWARN("Couldn't open tv zoom settings file %s", TV_ZOOM_SETTINGS_FILE);
+                    if (!settingsFile.Create())
+                    {
+                        LOGERR("Couldn't create tv zoom settings file %s", TV_ZOOM_SETTINGS_FILE);
+                        success = false;
+                    }
+                }
+
+                if (settingsFile.IsOpen())
+                {
+                    JsonObject settingsJson;
+                    settingsJson["zoomSetting"] = zoomSetting;
+                    if (!settingsJson.IElement::ToFile(settingsFile))
```

Comment:
We should persist the values for all platforms and not just for Platco/LLama where these new zoom settings are added.